src/codegen/replacements.ts

```diff
@@ -54,6 +54,11 @@ export const globalsToPrefix = [
   "undefined",
 ];
 
+replacements.push({
+  from: new RegExp(`\\bextends\\s+(${globalsToPrefix.join("|")})`, "g"),
+  to: "extends __no_intrinsic__%1",
```

Comment:
Is `__no_intrinsic__` a placeholder to swap things back or does it tie-into plucked references like a stored plucked reference to the `Promise` constructor?

Fun fact, the `Promise` that results from an `await` will not use `window.Promise` in non-Chromium environments and instead use the reference `%Promise%` which is the default value of `window.Promise`. So if you overwrite `window.Promise` it sill uses `%Promise%.

Reply:
its a workaround for the fact `--define Promise=__intrinsic__Promise` is in place.